modules/apps/portal/portal-http-impl/src/main/java/com/liferay/portal/http/internal/HttpImpl.java

```diff
@@ -1064,12 +1070,44 @@ private CloseableHttpClient _createProxyCloseableHttpClient() {
 		proxyHttpClientBuilder.setDefaultRequestConfig(
 			requestConfigBuilder.build());
 
+		proxyHttpClientBuilder.setKeepAliveStrategy(
+			_CONNECTION_KEEP_ALIVE_STRATEGY);
+
 		return proxyHttpClientBuilder.build();
 	}
 
+	private static final ConnectionKeepAliveStrategy
+		_CONNECTION_KEEP_ALIVE_STRATEGY = new ConnectionKeepAliveStrategy() {
+
+			@Override
+			public long getKeepAliveDuration(
+				HttpResponse httpResponse, HttpContext httpContext) {
+
+				long keepAliveDuration =
+					_defaultConnectionKeepAliveStrategy.getKeepAliveDuration(
+						httpResponse, httpContext);
+
+				if ((keepAliveDuration <= 0) &&
+					(_KEEPALIVE_DEFAULT_TIMEOUT > 0)) {
+
+					return _KEEPALIVE_DEFAULT_TIMEOUT * 1000L;
+				}
+
+				return keepAliveDuration;
+			}
+
+			private final ConnectionKeepAliveStrategy
+				_defaultConnectionKeepAliveStrategy =
+					new DefaultConnectionKeepAliveStrategy();
+
+		};
+
```

Comment:
@ericyanLr, it seems we only need to assign our ConnectionKeepAliveStrategy object when _KEEPALIVE_DEFAULT_TIMEOUT > 0. We may place the logic to the `activate()` as the belows:

```
@Activate
protected void activate() {
                  if (_KEEPALIVE_DEFAULT_TIMEOUT > 0) {
			_connectionKeepAliveStrategy = new ConnectionKeepAliveStrategy() {

				@Override
				public long getKeepAliveDuration(
					HttpResponse httpResponse, HttpContext httpContext) {

					if (keepAliveDuration <= 0) {
						return _KEEPALIVE_DEFAULT_TIMEOUT * 1000L;
					}
				}

				private final ConnectionKeepAliveStrategy
					_defaultConnectionKeepAliveStrategy =
						new DefaultConnectionKeepAliveStrategy();

			};
		}
}
private ConnectionKeepAliveStrategy _connectionKeepAliveStrategy;
```

For the usage `httpClientBuilder.setKeepAliveStrategy(_connectionKeepAliveStrategy);`，we can keep the current change. Because when _connectionKeepAliveStrategy is null, `httpClientBuilder.build()` will assign `DefaultConnectionKeepAliveStrategy.INSTANCE` to `MainClientExec#keepAliveStrategy`